uniffi_bindgen/src/templates/macros.rs

```diff
@@ -6,9 +6,14 @@
 {{ func.name() }}({% call _arg_list_rs_call(func) -%})
 {%- endmacro -%}
 
-{%- macro to_rs_call_with_prefix(prefix, func) -%}
+{%- macro to_rs_call_with_prefix(arg_name, func, obj) -%}
     {{ func.name() }}(
-    {{- prefix }}{% if func.arguments().len() > 0 %}, {% call _arg_list_rs_call(func) -%}{% endif -%}
+    {% if obj.threadsafe() %}
+    &*{{- arg_name }}
+    {% else %}
+    &mut *{{- arg_name }}.lock().unwrap()
+    {% endif %}
```

Comment:
This dereferencing/unwrapping logic...I wonder if it makes sense to move some or all of that into the definition of the `lift_rs` filter. It seems pretty analogous to me - this is the stuff that you have to do to go from the value that you get over the FFI, to the value that the Rust code needs to use at runtime.